annotation-utils/src/main/java/edu/isi/vista/annotationutils/ExtractAnnotationStats.kt

```diff
@@ -0,0 +1,223 @@
+                        tr {
+                            td {+"$totalAnnotations"}
+                            if (previousAnnotations == 0) {
+                                td{+"n/a"}
+                            }
+                            else {
+                                td { +"${totalAnnotations - previousAnnotations}" }
+                            }
+                        }
+                    }
+                    table {
+                        tr {
+                            th {+"User"}
+                            th {+"Annotations"}
+                            th {+"New annotations"}
+                        }
+                        for (user in sortedUsers.keys) {
+                            tr {
+                                td {+"$user"}
+                                td {+"${annotationsPerUser[user]}"}
+                                if (previousAnnotationsPerUser[user] == null) {
+                                    td {+"n/a"}
+                                }
+                                else {
+                                    td {+"${annotationsPerUser[user]?.minus(previousAnnotationsPerUser[user]!!)}"}
+                                }
+                            }
+                        }
+                    }
+                    table {
+                        tr {
+                            th {+"Subtype"}
+                            th {+"Annotations"}
+                            th {+"New annotations"}
+                        }
+                        for (subtype in sortedSubtypes.keys) {
+                            tr {
+                                td {+"$subtype" }
+                                td {+"${annotationsPerSubtype[subtype]}"}
+                                if (previousAnnotationsPerSubtype[subtype] == null) {
+                                    td {+"n/a"}
+                                }
+                                else {
+                                    td {+"${annotationsPerSubtype[subtype]?.minus(previousAnnotationsPerSubtype[subtype]!!)}"}
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            }
+        }
+        /**
+         * Search as far back as a week for a previous stats report
+         * The length of time can be changed
+         */
+        fun locatePreviousStats(currentDate: Calendar, statsDir: File): String {
+            val calendar = currentDate
+            val formatter = SimpleDateFormat("yyyy-MM-dd")
+            for (day in 1..7) {
+                calendar.add(Calendar.DAY_OF_YEAR, -1)
+                val lastDate = formatter.format(calendar.getTime())
+                val lastFilename = "$statsDir/StatsReport$lastDate.html"
+                if (File(lastFilename).exists()) {
+                    return lastFilename
+                }
+            }
+            return ""
```

Comment:
Avoid using the empty string to represent things which are missing. Prefer to return `Optional<String>` so that the type system can make sure the user takes into account the possibility that the returned value might be missing.

Reply:
You might actually want to return a data class object here so you could also bundle in the date. That way your HTML can tell how indicate when the snapshot it is comparing against was.